R/checkout.R

```diff
@@ -2,21 +2,48 @@
 #' Checkout a Repository
 #'
 #' `renv::checkout()` can be used to install the latest packages available from
-#' the requested repositories.
+#' a (set of) package repositories.
+#'
+#' `renv::checkout()` is most useful with services like the Posit's
+#' [Package Manager](https://packagemanager.rstudio.com/), as it
+#' can be used to switch between different repository snapshots within an
+#' `renv` project. In this way, you can upgrade (or downgrade) all of the
+#' packages used in a particular `renv` project to the package versions
+#' provided by a particular snapshot.
 #'
 #' @inheritParams renv-params
 #'
-#' @param repos The \R package repositories to check out. When `NULL` (the default),
-#'   `renv` will attempt to determine the packages used in the project via
-#'   a call to [renv::dependencies()].
+#' @param repos The \R package repositories to check out.
 #'
 #' @param packages The packages to be installed. When `NULL` (the default),
-#'   all packages currently used in the project will be installed.
+#'   all packages currently used in the project will be installed, as
+#'   determined by [renv::dependencies()]. The recursive dependencies of these
+#'   packages will be included as well.
+#'
+#' @param date The snapshot date to use. When set, the associated snapshot as
+#'   available from the Posit's public
+#'   [Package Manager](https://packagemanager.rstudio.com/) instance will be
+#'   used. The repository path can be adjusted using the `renv.checkout.repos`
+#'   option. Ignored if `repos` is non-`NULL`.
+#'
+#' @examples
+#' \dontrun{
+#'
+#' # check out packages from PPM using the date '2023-01-02'
+#' renv::checkout(date = "2023-01-02")
+#'
+#' # alternatively, supply the full repository path
+#' renv::checkout(repos = "https://packagemanager.rstudio.com/cran/2023-01-02")
+#'
+#' # only check out some subset of packages (and their recursive dependencies)
+#' renv::checkout(packages = "dplyr", date = "2023-01-02")
 #'
-#' @keywords internal
-checkout <- function(repos = getOption("repos"),
+#' }
+#' @export
+checkout <- function(repos = NULL,
                      ...,
                      packages = NULL,
+                     date     = NULL,
```

Comment:
Given it's importance, I think `date` should come earlier in the arguments.

Reply:
Makes sense -- I use `...` here to force arguments to be named; should I relax that requirement (remove the usage of `...`)?